### PR TITLE
Ai branch

### PR DESCRIPTION
**Pull request by Tomerd-afk**
I now have: 

A Prefab for Cyborgs with Basic rifles that can follow the player and shoot their guns
I have an EnemyAI and an ObjectSpawnerPool that pools enemy AI and allows them to act like boids.

### CODE REVIEW

File: Assets/Prefabs/EnemyMid .prefab [hunk 1]
```diff
@@ -0,0 +1,132 @@
+%YAML 1.1
+%TAG !u! tag:unity3d.com,2011:
```
Comment (toryjleo) [inline Assets/Prefabs/EnemyMid .prefab:2]: Please remove the whitespace at the end of this filename

File: Assets/Scripts/AI/EnemyAI.cs [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+    public float attackRange; 
+
+    bool alive;
+
+
+    private void Awake()
+    {
+        alive = true;
+        rb = GetComponent<Rigidbody>();
+        location = transform.position;
```
Comment (toryjleo) [inline Assets/Scripts/AI/EnemyAI.cs:25]: Can you use transform,position?

File: Assets/Scripts/AI/EnemyAI.cs [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+
+        targetVec = target.transform.position;
+        location = transform.position;
+        //seek(targetVec);
+        arrive(targetVec);
+        //wander();
+           
+    }
+
+    public void seperate(List<EnemyAI> pool) //this function will edit the steer of an AI so it moves away from nearby other AI
```
Comment (toryjleo) [inline Assets/Scripts/AI/EnemyAI.cs:45]: Upper case method name

Comment (toryjleo) [reply Assets/Scripts/AI/EnemyAI.cs:45]: Also can you rename the variable to something like aiPool? It makes me think you are passing in an objectpool object

File: Assets/Scripts/AI/EnemyAI.cs [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+    {
+
+        targetVec = target.transform.position;
+        location = transform.position;
+        //seek(targetVec);
+        arrive(targetVec);
+        //wander();
+           
+    }
+
```
Comment (toryjleo) [inline Assets/Scripts/AI/EnemyAI.cs:44]:
```suggestion
/// <summary> this function will edit the steer of an AI so it moves away from nearby other AI</summary>
<param name="pool">description</param>
```

File: Assets/Scripts/AI/EnemyAI.cs [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+            }
+            
+        }
+    }
+
+    public bool isAlive()
+    {
+        return alive;
+    }
+
```
Comment (toryjleo) [inline Assets/Scripts/AI/EnemyAI.cs:90]:
```suggestion
/// <summary> sets the target of the entity</summary>
<param name="targ">description</param>
```

File: Assets/Scripts/AI/EnemyAI.cs [hunk 6]
```diff
@@ -0,0 +1,173 @@
...
+        return location;
+    }
+
+    void applyForce(Vector3 force)
+    {
+        rb.AddForce(force);
+        
+    }
+
+    private void seek(Vector3 target)
```
Comment (toryjleo) [inline Assets/Scripts/AI/EnemyAI.cs:110]: Ditto previous documentation comments

File: Assets/Scripts/AI/EnemyAI.cs [hunk 7]
```diff
@@ -0,0 +1,173 @@
...
+
+    private Vector3 targetVec; //this is the vector to their quarry 
+    private Vector3 location; //this is the vector of the entity 
+    private Vector3 forward;
+    public GameObject target;
+    public Rigidbody rb;
+    public Gun myGun; 
+
+    float maxSpeed;
+    public float attackRange;
```
Comment (toryjleo) [inline Assets/Scripts/AI/EnemyAI.cs:16]: Where is this set????

File: Assets/Scripts/AI/EnemyAI.cs [hunk 8]
```diff
@@ -0,0 +1,173 @@
...
+        desiredVec.Normalize();
+        desiredVec *= maxSpeed;
+
+        Vector3 steer = desiredVec - rb.velocity;
+        
+        //steer.limit(maxForce);
+        applyForce(steer); 
+    } // simpler function to just chase directly after the player 
+
+    private void arrive(Vector3 target) //This can be used for Enemies that stay at range and dont run into melee.
```
Comment (toryjleo) [inline Assets/Scripts/AI/EnemyAI.cs:128]: Ditto previous documentation comments

File: Assets/Scripts/AI/EnemyAI.cs [hunk 9]
```diff
@@ -0,0 +1,173 @@
...
+        //steer.limit(maxForce);
+
+        this.myGun.Shoot(target); 
+
+        applyForce(steer);
+    } 
+
+    private void wander() //do this later 
+    {
+
```
Comment (toryjleo) [inline Assets/Scripts/AI/EnemyAI.cs:156]:
```suggestion
\\ TODO: <Whatever you need to do later>
```

File: Assets/Scripts/ObjectPoolSpawner.cs [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+    public Gun gunToPool;
+    public GameObject player; 
+    private List<EnemyAI> pool;
+    public float size;
+    
+
+    // Start is called before the first frame update
+    void Start()
+    {
+        pool = new List<EnemyAI>();
```
Comment (toryjleo) [inline Assets/Scripts/ObjectPoolSpawner.cs:18]: Maybe move this to an Init() method?

File: Assets/Scripts/AI/EnemyAI.cs (hunk 3 above)
Comment (Tomerd-afk) [reply Assets/Scripts/AI/EnemyAI.cs:45]: FUCK YOU TORY, I'm going to make this class EVEN MORE GENERIC IN THE FUTURE

File: Assets/Scripts/ObjectPoolSpawner.cs (hunk 10 above)
Comment (Tomerd-afk) [reply Assets/Scripts/ObjectPoolSpawner.cs:18]: Why? The Pool is going to be a scene object and doesn't need to be initialized by anything else?

**Review by Tomerd-afk — COMMENTED** [review]
All in all great

File: Assets/Scripts/ObjectPoolSpawner.cs (hunk 10 above)
Comment (toryjleo) [reply Assets/Scripts/ObjectPoolSpawner.cs:18]: I mean you can keep it however I try to shy away from having a bunch of garbage in Start()

Comment (Tomerd-afk) [reply Assets/Scripts/ObjectPoolSpawner.cs:18]: I added a TODO and will update later down the line if nessicary

**Review by Tomerd-afk — COMMENTED** [review]
ADDED MORE CHANGES

**Pull request merged**